### PR TITLE
add support for configurable remote transport deadline

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
add support for configurable remote transport deadline

## Motivation and Context
configurable remote transport timeouts for some special cases
where this value needs to be bumped to a higher value when
transferring large data between federated instances.

## How to test this PR?
Nothing special

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [ ] Unit tests needed

### CODE REVIEW

File: cmd/object-handlers.go [hunk 1]
```diff
@@ -711,16 +711,16 @@ func getCpObjMetadataFromHeader(ctx context.Context, r *http.Request, userMeta m
...
+	getRemoteInstanceTransportOnce sync.Once
+)
 
 // Returns a minio-go Client configured to access remote host described by destDNSRecord
 // Applicable only in a federated deployment
 var getRemoteInstanceClient = func(r *http.Request, host string) (*miniogo.Core, error) {
 	getRemoteInstanceTransportOnce.Do(func() {
-		getRemoteInstanceTransport = NewGatewayHTTPTransport()
-		getRemoteInstanceTransportLongTO = newGatewayHTTPTransport(time.Hour)
+		getRemoteInstanceTransport = newGatewayHTTPTransport(2 * time.Hour)
```
Comment (klauspost) [inline cmd/object-handlers.go]: Is this the value you wanted to make configurable?

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :x: [more...](#b1937b53-192e-455c-865e-4c52f3b8fd5a) |
| mint-gateway-azure.sh | :x: [more...](#ba945d3b-dbe3-4e4b-9f8c-27d9a1efa21a) |
| mint-gateway-nas.sh | :x: [more...](#97e311cb-85ee-4d62-a43e-af637ed3d38f) |
<a name="b1937b53-192e-455c-865e-4c52f3b8fd5a"></a>
### 10447-21832bc/mint-gateway-s3.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:31162
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp f80e5449f1a0:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 2 seconds
(5/15) Running awscli tests ... done in 1 minutes and 10 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 23 seconds
(8/15) Running minio-dotnet tests ... done in 33 seconds
(9/15) Running minio-go tests ... FAILED in 16 minutes and 3 seconds
{
  "args": {
    "bucketName": "minio-go-test-elv0r50m5n1dz329",
    "objectName": "minio-go-test-elv0r50m5n1dz329unique",
    "opts": "",
    "size": 135266304
  },
  "duration": 956317,
  "error": "The specified bucket does not exist",
  "function": "PutObject(bucketName, objectName, reader, size, opts)",
  "message": "PutObjectWithSize failed",
  "name": "minio-go: testPutObjectNoLengthV2",
  "status": "FAIL"
}
(9/15) Running minio-java tests ... done in 42 seconds
(10/15) Running minio-js tests ... done in 36 seconds
(11/15) Running minio-py tests ... done in 1 minutes and 55 seconds
(12/15) Running s3cmd tests ... done in 1 minutes and 4 seconds
(13/15) Running s3select tests ... done in 4 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
<a name="ba945d3b-dbe3-4e4b-9f8c-27d9a1efa21a"></a>
### 10447-21832bc/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev4.minio.io:31236
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 61996b5b869f:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 8 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 1 minutes and 6 seconds
(4/15) Running aws-sdk-ruby tests ... done in 17 seconds
(5/15) Running awscli tests ... FAILED in 43 seconds
{
  "name": "awscli",
  "duration": 3693,
  "function": "aws --endpoint-url http://minio-dev4.minio.io:31236 s3api copy-object --bucket awscli-mint-test-bucket-17930 --key datafile-1-kB-copy --copy-source awscli-mint-test-bucket-17930/datafile-1-kB\n",
  "status": "FAIL",
  "error": "Hash mismatch expected a861ac626e0bee4421a50737c581233c, got 0x8D855683D203877-1"
}
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 3 minutes and 20 seconds
(7/15) Running minio-dotnet tests ... done in 1 minutes and 34 seconds
(8/15) Running minio-go tests ... FAILED in 1 minutes and 50 seconds
{
  "args": {
    "bucketName": "minio-go-test-ybebsu0fa1rwp5o5",
    "objectName": "test-object",
    "opts": "",
    "size": -1
  },
  "duration": 715,
  "function": "PutObject(bucketName, objectName, reader,size,opts)",
  "message": "Unexpected size",
  "name": "minio-go: testPutObjectStreaming",
  "status": "FAIL"
}
(8/15) Running minio-java tests ... done in 7 minutes and 26 seconds
(9/15) Running minio-js tests ... FAILED in 44 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 91,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:86:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:678:10) at endReadableNT (_stream_readable.js:1244:12) at processTicksAndRejections (internal/process/task_queues.js:80:21)"
}
(9/15) Running minio-py tests ... done in 13 minutes and 4 seconds
(10/15) Running s3cmd tests ... done in 2 minutes and 14 seconds
(11/15) Running s3select tests ... done in 33 seconds
(12/15) Running security tests ... done in 0 seconds

Executed 12 out of 15 tests successfully.
```
<a name="97e311cb-85ee-4d62-a43e-af637ed3d38f"></a>
### 10447-21832bc/mint-gateway-nas.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:31963
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp b35f32519f8c:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 0 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 41 seconds
(4/15) Running aws-sdk-ruby tests ... done in 2 seconds
(5/15) Running awscli tests ... FAILED in 33 seconds
{
  "name": "awscli",
  "duration": 16965,
  "function": "aws --endpoint-url http://minio-c3.minio.io:31963 s3api get-object --bucket awscli-mint-test-bucket-1926 --key awscli-mint-test-bucket-1926-object /tmp/datafile-0-b\n",
  "status": "FAIL",
  "error": "An error occurred (NoSuchBucket) when calling the GetObject operation: The specified bucket does not exist"
}
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 22 seconds
(7/15) Running minio-dotnet tests ... done in 31 seconds
(8/15) Running minio-go tests ... done in 47 seconds
(9/15) Running minio-java tests ... done in 20 seconds
(10/15) Running minio-js tests ... done in 32 seconds
(11/15) Running minio-py tests ... done in 1 minutes and 10 seconds
(12/15) Running s3cmd tests ... done in 23 seconds
(13/15) Running s3select tests ... done in 3 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally
Error: No such image: minio/minio:10447-21832bc

File: cmd/object-handlers.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/object-handlers.go]: Yes it is configured in config-current.go @klauspost

Comment (klauspost) [reply cmd/object-handlers.go]: @harshavardhana Shouldn't it be read from there, then? This is a constant.

Comment (harshavardhana) [reply cmd/object-handlers.go]: Correct I added it here for safety @klauspost since it uses a really high value

Comment (harshavardhana) [reply cmd/object-handlers.go]: Let me think If I can remove and change it further.

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#2b7691e1-c830-4f2c-a109-6166679c8398) |
<a name="2b7691e1-c830-4f2c-a109-6166679c8398"></a>
### 10447-a7dc2e6/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:32083
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 3487ddb388bd:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 8 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 1 minutes and 11 seconds
(4/15) Running aws-sdk-ruby tests ... done in 18 seconds
(5/15) Running awscli tests ... FAILED in 45 seconds
{
  "name": "awscli",
  "duration": 4059,
  "function": "aws --endpoint-url http://minio-c3.minio.io:32083 s3api copy-object --bucket awscli-mint-test-bucket-10424 --key datafile-1-kB-copy --copy-source awscli-mint-test-bucket-10424/datafile-1-kB\n",
  "status": "FAIL",
  "error": "Hash mismatch expected a861ac626e0bee4421a50737c581233c, got 0x8D855B877076ED5-1"
}
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 6 minutes and 48 seconds
(7/15) Running minio-dotnet tests ... done in 1 minutes and 33 seconds
(8/15) Running minio-go tests ... FAILED in 1 minutes and 56 seconds
{
  "args": {
    "bucketName": "minio-go-test-pzqc2kbk240org4y",
    "objectName": "test-object",
    "opts": "",
    "size": -1
  },
  "duration": 732,
  "function": "PutObject(bucketName, objectName, reader,size,opts)",
  "message": "Unexpected size",
  "name": "minio-go: testPutObjectStreaming",
  "status": "FAIL"
}
(8/15) Running minio-java tests ... done in 7 minutes and 44 seconds
(9/15) Running minio-js tests ... FAILED in 45 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 81,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:86:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:678:10) at endReadableNT (_stream_readable.js:1244:12) at processTicksAndRejections (internal/process/task_queues.js:80:21)"
}
(9/15) Running minio-py tests ... done in 14 minutes and 30 seconds
(10/15) Running s3cmd tests ... done in 2 minutes and 9 seconds
(11/15) Running s3select tests ... done in 36 seconds
(12/15) Running security tests ... done in 0 seconds

Executed 12 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally
Error: No such image: minio/minio:10447-a7dc2e6

**Comment by harshavardhana** [comment]
PTAL @klauspost @vadmeste

**Review by klauspost — APPROVED** [review]
LGTM

File: cmd/config/api/help.go [hunk 2]
```diff
@@ -39,5 +39,11 @@ var (
 			Optional:    true,
 			Type:        "csv",
 		},
+		config.HelpKV{
+			Key:         apiRemoteTransportDeadline,
+			Description: `set the deadline for API requests on remote transports while proxying between federated instances e.g. "2h"`,
```
Comment (vadmeste) [inline cmd/config/api/help.go:44]: I think 'remote transports' keywords are not clear for users.

```
set the deadline for API requests while proxying between federated instances
```

File: cmd/config/api/api.go [hunk 3]
```diff
@@ -59,15 +61,20 @@ var (
...
 type Config struct {
-	APIRequestsMax      int           `json:"requests_max"`
-	APIRequestsDeadline time.Duration `json:"requests_deadline"`
-	APIReadyDeadline    time.Duration `json:"ready_deadline"`
-	APICorsAllowOrigin  []string      `json:"cors_allow_origin"`
+	RequestsMax             int           `json:"requests_max"`
+	RequestsDeadline        time.Duration `json:"requests_deadline"`
+	ReadyDeadline           time.Duration `json:"ready_deadline"`
+	CorsAllowOrigin         []string      `json:"cors_allow_origin"`
+	RemoteTransportDeadline time.Duration `json:"remote_transport_deadline"`
```
Comment (vadmeste) [inline cmd/config/api/api.go:77]: maybe RemoteFederatedDeadline is a better name ?

**Review by vadmeste — COMMENTED** [review]
Some comments

File: cmd/config/api/api.go (hunk 3 above)
Comment (harshavardhana) [reply cmd/config/api/api.go:77]: I want to keep it generic we may remove federation in future..

File: cmd/config/api/help.go (hunk 2 above)
Comment (harshavardhana) [reply cmd/config/api/help.go:44]: It is technically correct, since the usage of this is rare and niche - I think we are okay to call it this way..

**Review by kannappanr — APPROVED** [review]
LGTM

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#098f3bec-064f-4c0d-a0d7-e3ac39cc4d26) |
| mint-dist-erasure.sh | :x: [more...](#cd61896b-9e1e-4bde-aa2c-04bb877a1fe9) |
<a name="098f3bec-064f-4c0d-a0d7-e3ac39cc4d26"></a>
### 10447-59cc9a1/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:32051
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp f2b9d7732f59:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 8 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 1 minutes and 23 seconds
(4/15) Running aws-sdk-ruby tests ... done in 21 seconds
(5/15) Running awscli tests ... FAILED in 43 seconds
{
  "name": "awscli",
  "duration": 4053,
  "function": "aws --endpoint-url http://minio-dev7.minio.io:32051 s3api copy-object --bucket awscli-mint-test-bucket-20392 --key datafile-1-kB-copy --copy-source awscli-mint-test-bucket-20392/datafile-1-kB\n",
  "status": "FAIL",
  "error": "Hash mismatch expected a861ac626e0bee4421a50737c581233c, got 0x8D856E33BF49880-1"
}
(5/15) Running healthcheck tests ... done in 1 seconds
(6/15) Running mc tests ... done in 3 minutes and 49 seconds
(7/15) Running minio-dotnet tests ... done in 1 minutes and 34 seconds
(8/15) Running minio-go tests ... done in 6 minutes and 18 seconds
(9/15) Running minio-java tests ... FAILED in 38 seconds
{
  "name": "minio-java",
  "function": "uploadObject()",
  "args": "[multi-part upload]",
  "duration": 846,
  "status": "FAIL",
  "error": "java.net.SocketException: Connection reset >>> [java.base/java.net.SocketInputStream.read(SocketInputStream.java:186), java.base/java.net.SocketInputStream.read(SocketInputStream.java:140), okio.Okio$2.read(Okio.java:140), okio.AsyncTimeout$2.read(AsyncTimeout.java:237), okio.RealBufferedSource.indexOf(RealBufferedSource.java:358), okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.java:230), okhttp3.internal.http1.Http1ExchangeCodec.readHeaderLine(Http1ExchangeCodec.java:242), okhttp3.internal.http1.Http1ExchangeCodec.readResponseHeaders(Http1ExchangeCodec.java:213), okhttp3.internal.connection.Exchange.readResponseHeaders(Exchange.java:115), okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.java:94), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142), okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:43), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:117), okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:94), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:117), okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142), okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:88), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:117), okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:229), okhttp3.RealCall.execute(RealCall.java:81), io.minio.MinioClient.execute(MinioClient.java:1014), io.minio.MinioClient.uploadPart(MinioClient.java:7781), io.minio.MinioClient.putObject(MinioClient.java:4706), io.minio.MinioClient.uploadObject(MinioClient.java:4931), FunctionalTest.testUploadObject(FunctionalTest.java:706), FunctionalTest.uploadObject_test(FunctionalTest.java:730), FunctionalTest.runTests(FunctionalTest.java:3791), FunctionalTest.main(FunctionalTest.java:4056)]"
}
(9/15) Running minio-js tests ... done in 2 minutes and 44 seconds
(10/15) Running minio-py tests ... done in 13 minutes and 2 seconds
(11/15) Running s3cmd tests ... done in 2 minutes and 31 seconds
(12/15) Running s3select tests ... done in 1 minutes and 3 seconds
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
<a name="cd61896b-9e1e-4bde-aa2c-04bb877a1fe9"></a>
### 10447-59cc9a1/mint-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev8.minio.io:31390
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 84814e0ba552:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 4 seconds
(5/15) Running awscli tests ... done in 1 minutes and 22 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 29 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 7 seconds
(9/15) Running minio-go tests ... FAILED in 1 hours 12 minutes and 44 seconds
{
  "args": {
    "bucketName": "minio-go-test-df9kbqj6b5xw432v",
    "fileName": "/mint/data/datafile-129-MB",
    "location": "us-east-1",
    "objectName": "minio-go-test-df9kbqj6b5xw432vFPutObject-GTar",
    "opts": {
      "UserMetadata": null,
      "UserTags": null,
      "Progress": null,
      "ContentType": "",
      "ContentEncoding": "",
      "ContentDisposition": "",
      "ContentLanguage": "",
      "CacheControl": "",
      "Mode": "",
      "RetainUntilDate": "0001-01-01T00:00:00Z",
      "ServerSideEncryption": null,
      "NumThreads": 0,
      "StorageClass": "",
      "WebsiteRedirectLocation": "",
      "PartSize": 0,
      "LegalHold": "",
      "SendContentMd5": false,
      "DisableMultipart": false,
      "ReplicationVersionID": "",
      "ReplicationETag": "",
      "ReplicationStatus": "",
      "ReplicationMTime": "0001-01-01T00:00:00Z"
    }
  },
  "duration": 4290508,
  "error": "A timeout occurred while trying to lock a resource, please reduce your request rate",
  "function": "FPutObject(bucketName, objectName, fileName, opts)",
  "message": "FPutObject failed",
  "name": "minio-go: testFPutObject",
  "status": "FAIL"
}
(9/15) Running minio-java tests ... done in 54 seconds
(10/15) Running minio-js tests ... done in 48 seconds
(11/15) Running minio-py tests ... done in 2 minutes and 35 seconds
(12/15) Running s3cmd tests ... done in 23 seconds
(13/15) Running s3select tests ... done in 5 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally
Error: No such image: minio/minio:10447-59cc9a1

**Pull request merged**